src/VideoNativeComponent.ts

```diff
@@ -0,0 +1,323 @@
+
+type Drm = Readonly<{
+  drmType?: DrmType;
+  licenseServer?: string;
+  headers?: Headers;
+  contentId?: string; // ios
+  certificateUrl?: string; // ios
+  base64Certificate?: boolean; // ios default: false
+  useExternalGetLicense?: boolean; // ios
+}>
+
+type TextTracks = ReadonlyArray<Readonly<{
+  title: string;
+  language: string;
+  type: string;
+  uri: string;
+}>>
+
+type TextTrackType = 'system' | 'disabled' | 'title' | 'language' | 'index';
+
+type SelectedTextTrack = Readonly<{
+  selectedTextType?: TextTrackType;
+  value?: string;
+  index?: number;
+}>
+
+type AudioTrackType = 'system' | 'disabled' | 'title' | 'language' | 'index';
+
+type SelectedAudioTrack = Readonly<{
+  selectedAudioType?: AudioTrackType;
+  value?: string;
+  index?: number;
+}>
+
+export type Seek = Readonly<{
+  time: number;
+  tolerance?: number;
+}>
+
+type BufferConfig = Readonly<{
+  minBufferMs?: number;
+  maxBufferMs?: number;
+  bufferForPlaybackMs?: number;
+  bufferForPlaybackAfterRebufferMs?: number;
+  maxHeapAllocationPercent?: number;
+  minBackBufferMemoryReservePercent?: number;
+  minBufferMemoryReservePercent?: number;
+}>
+
+type SelectedVideoTrack = Readonly<{
+  type?: 'auto' | 'disabled' | 'resolution' | 'index';
+  value?: number;
+}>
+
+type SubtitleStyle = Readonly<{
+  fontSize?: number;
+  paddingTop?: number;
+  paddingBottom?: number;
+  paddingLeft?: number;
+  paddingRight?: number;
+}>
+
+export type OnLoadData = Readonly<{
+  currentTime: number;
+  duration: number;
+  naturalSize: Readonly<{
+    width: number;
+    height: number;
+    orientation: string;
+  }>;
+}>
+
+export type OnLoadStartData = Readonly<{
+  isNetwork: boolean;
+  type: string;
+  uri: string;
+}>
+
+export type OnBufferData = Readonly<{ isBuffering: boolean }>;
+
+
+export type OnProgressData = Readonly<{
+  currentTime: number;
+  playableDuration: number;
+  seekableDuration: number;
+}>
+
+export type OnBandwidthUpdateData = Readonly<{
+  bitrate: number;
+}>;
+
+export type OnSeekData = Readonly<{
+  currentTime: number;
+  seekTime: number;
+  finished: boolean;
+}>
+
+export type OnPlaybackStateChangedData = Readonly<{
+  isPlaying: boolean;
+}>
+
+export type OnTimedMetadataData = Readonly<{
+  metadata: ReadonlyArray<Readonly<{
+    value?: string
+    identifier?: string
+  }>>
+}>
+
+
+export type OnAudioTracksData = Readonly<{
+  audioTracks: ReadonlyArray<Readonly<{
+    index?: number
+    title?: string
+    language?: string
+    bitrate?: number
+    type?: string
+    selected?: boolean
+  }>>
+}>
+
+export type OnTextTracksData = Readonly<{
+  textTracks: ReadonlyArray<Readonly<{
+    index?: number
+    title?: string
+    language?: string
+    /**
+     * iOS only supports VTT, Android supports all 3
+     */
+    type?: 'srt' | 'ttml' | 'vtt'
+    selected?: boolean
+  }>>
+}>
+
+export type OnVideoTracksData = Readonly<{
+  videoTracks: ReadonlyArray<Readonly<{
+    trackId?: number
```

Comment:
Track id is mandatory I think you can remove the ?

Reply:
Fixed ✅